src/main/groovy/com/morpheusdata/xen/XenserverProvisionProvider.groovy

```diff
@@ -1401,20 +1400,42 @@ class XenserverProvisionProvider extends AbstractProvisionProvider implements Wo
 	@Override
 	ServiceResponse resizeWorkload(Instance instance, Workload workload, ResizeRequest resizeRequest, Map opts) {
 		log.debug("resizeWorkload workload?.id: ${workload?.id} - opts: ${opts} - workload.id: ${workload.id}")
-		ServiceResponse rtn = ServiceResponse.success()
+		log.info("resizeWorkload calling resizeWorkloadAndHost")
+		return resizeWorkloadAndHost(instance?.id, workload, workload.server, resizeRequest, opts, true)
+	}
 
-		ComputeServer computeServer = context.async.computeServer.get(workload.server.id).blockingGet()
+	@Override
+	ServiceResponse resizeServer(ComputeServer server, ResizeRequest resizeRequest, Map opts) {
+		log.info("resizeServer calling resizeWorkloadAndHost")
+		return resizeWorkloadAndHost(null, null, server, resizeRequest, opts, false)
+	}
+
+	private ServiceResponse resizeWorkloadAndHost (Long instanceId, Workload workload, ComputeServer server, ResizeRequest resizeRequest, Map opts, Boolean isWorkload) {
```

Comment:
Rename this to resizeWorkloadAndServer as it is used in resizeWorkload and resizeServer methods and a Host is a term used for something else in Morpheus.

Reply:
change added.